lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
It is not closed and we can still write to this socket (hanging or not hanging).

I do not like approach with this timeout because it is not user friendly. Let's think, how it can be used for 408, frankly speaking it is beyond the scope of this PR:

Reply with 408 is now not a simple task, and I think a great number of node users can't handle 408 right because of this complexity, for example, user code:
```JavaScript
server.setTimeout(8000, function(socket) {
  socket.end('we have a timeout event here');
  socket.destroy();
});
```
will write to still open TCP socket and client/brows will receive this data in both cases: if connection is ended successfully and if connection is ```hanging```.
So, how user can detect is it hanging:
```JavaScript
server.setTimeout(8000, function(socket) {
  var res = socket._httpMessage;
  if (res) {
    res.statusCode = 408;
    res.end('HTTP 408 Request Timeout');
  }
  socket.destroy();
});
```
Not a big deal, but ```socket._httpMessage``` is not obvious and well documented. But this code is bad, because ```socket.destroy();``` may destroy socket before it send 408 to client/browser. In test case it works, but under high load I tested that and received no 408 in about 27% (percentage is not very important, because we are interested in a fact of such possibility).
```JavaScript
server.setTimeout(2000, function(socket) {
  var res = socket._httpMessage;
  if (res) {
    res.statusCode = 408;
    res.end('HTTP 408 Request Timeout');
    res.on('finish', function() {
      socket.destroy();
    });
  } else socket.destroy();
});
```
I am afraid that it is too difficult for such a simple case. It would be great to have something like this in user code:
```JavaScript
server.setTimeout(2000, function(socket, res) {
  res.statusCode = 408;
  res.end('HTTP 408 Request Timeout');
  socket.destroySoon();
});
```